plugins/modules/parcel.py

```diff
@@ -0,0 +1,224 @@
+)
+
+from cm_client import ClustersResourceApi, ParcelResourceApi
+from cm_client.rest import ApiException
+import time
+
+ANSIBLE_METADATA = {
+    "metadata_version": "1.1",
+    "status": ["preview"],
+    "supported_by": "community",
+}
+
+DOCUMENTATION = r"""
+---
+module: parcel
+short_description: Manage the state of parcels on a Cluster
+description:
+  - Facilitates the management of parcels on a Cluster by downloading, distributing, and activating them according to the specified state.
+  - Supported states include 'download', 'distribute', and 'activate', each corresponding to specific actions performed on parcels.
+author:
+  - "Ronald Suplina (@rsuplina)"
+requirements:
+  - cm_client
+"""
+
+EXAMPLES = r"""
+---
+- name: Download, distribute and activate a parcel on a cluster 
+  cloudera.cluster.parcel:
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    cluster_name: "OneNodeECS"
+    product: "ECS"
+    parcel_version: "1.5.1-b626-ecs-1.5.1-b626.p0.42068229"
+    state: "activate"
+
+- name: Downloand and distribute a parcel on a cluster 
+  cloudera.cluster.parcel:
+    host: example.cloudera.com
+    username: "jane_smith"
+    password: "S&peR4Ec*re"
+    cluster_name: "OneNodeECS"
+    product: "ECS"
+    parcel_version: "1.5.1-b626-ecs-1.5.1-b626.p0.42068229"
+    state: "distribute"
+
+"""
+
+RETURN = r"""
+---
+cloudera_manager:
+    description: Returns details about specific parcel
+    type: dict
+    contains:
+        product:
+            product: The name of the product.
+            type: str
+            returned: optional
+        version:
+            description: The version of the product
+            type: str
+            returned: optional
+        stage:
+            description: Current stage of the parcel.
+            type: str
+            returned: optional
+        state:
+            description: The state of the parcel. This shows the progress of state transitions and if there were any errors.
+            type: dict
+            returned: optional
+        clusterRef:
+            description:  A reference to the enclosing cluster.
+            type: dict
+            returned: optional
+        displayName:
+            description: Display name of the parcel.
+            type: str
+            returned: optional
+        description:
+            description: Description of the parcel.
+            type: str
+            returned: optional
+"""
+
+
+class ClouderaParcel(ClouderaManagerModule):
+    def __init__(self, module):
+        super(ClouderaParcel, self).__init__(module)
+
+        self.cluster_name = self.get_param("cluster_name")
+        self.product = self.get_param("product")
+        self.parcel_version = self.get_param("parcel_version")
+        self.state = self.get_param("state")
+
+        self.process()
+
+
+
+    def download_parcel(self, parcel_api_instance, cluster_name, product, parcel_version, polling_interval):
```

Comment:
Need to break out of the `while` loop on error with `fail_json`. Should also put a good timeout guard as well. (But with a bunch of time, as some states take a while.)